transport/internal/errors/peers.go

```diff
@@ -94,3 +94,43 @@ type ErrInvalidAgentConversion struct {
 func (e ErrInvalidAgentConversion) Error() string {
 	return fmt.Sprintf("cannot convert agent (%v) to type %s", e.Agent, e.ExpectedType)
 }
+
+// ErrNoPeerToSelect is used when a peerlist doesn't have any peers to return
+type ErrNoPeerToSelect string
```

Comment:
Note to replace this with a ErrNoPeerTimeout when Choose blocking. This error should never occur in any peer list implementation, since a peer list must assume that the peer provider may eventually provide an identifier, and the agent may eventually provide a peer.